### DISCUSSION TITLE
Real time stopped working

### DISCUSSION THREAD

**Discussion by Blackshuck4** [fra589/grbl-Mega-5X#231] [Q&A] [answered]
Hi Fra589, Brilliant programme. Took me a while, but now I have a fully working 4th axis rotary, with homing (ArduinoMega5X). BUT, for some bizarre reason, recently, when I switch from simulation mode to my machine (I'm using the Shapeoko 3/4 axis in machine manager), all the readouts and the machine simulation don't move/update. In simulation mode everything is perfect. In machine mode I can see the G-code flying by in the control panels, and my actual machine is doing all the moves. Jogging and homing are also fine, it's just that the coordinate readouts and model aren't reflecting any machine movements.
Any ideas where the problem might lie?

**Comment by fra589** [comment]
Hi @Blackshuck4,  
  
To know what can happen, I need some more information...
What is your version of grbl-Mega-5X? (result of the $I command)
What is your grbl configuration? (result of the $$ command)
What is the result of "?" command when the machine is in normal operation? 
What software do you use to drive grbl-Mega-5X?
It may be a problem in the configuration of your grbl sender and not of grbl itself.
   
@++;  
Gauthier.

**Reply by Blackshuck4** [reply]
Hi Gauthier, thanks so much for getting back to me so quickly!
OK:
version of grbl-Mega-5X? (result of the $I command)
grb[VER:1.2d.20210930:]
[AXS:4:XYZA]
[OPT:VNMGH,35,255,28]

grbl configuration? (result of the $$ command):
$0=10
$1=255
$2=0
$3=6
$4=1
$5=0
$6=0
$10=0
$11=0.020
$12=0.002
$13=0
$20=0
$21=1
$22=1
$23=7
$24=100.000
$25=500.000
$26=250
$27=5.000
$30=12000
$31=550
$32=0
$100=294.000
$101=294.000
$102=294.000
$103=17.800
$110=750.000
$111=750.000
$112=750.000
$113=750.000
$120=50.000
$121=50.000
$122=50.000
$123=50.000
$130=400.000
$131=200.000
$132=200.000
$133=360.000
ok

Result of "?" command when the machine is in normal operation?
I just get 
Grbl 1.2d ['$' for help]
 ok in RHS control panel

What software do you use to drive grbl-Mega-5X?
I'm using Grblgru v4.2.0

Thanks for your help,

**Comment by fra589** [comment, marked as answer]
Hi @Blackshuck4,  
  
I don't know what is "RHS control panel" ?  
  
I don't know about Grblgru since I working under Debian Linux and I have [my own Grbl's control panel / gcode sender](https://github.com/fra589/cn5X).  
I seams than GrblGru had a newer version 4.5 than your 4.2... Perhaps you should update?  
  
Reading some info on the web, I see that simulation mode is a specific to the GrblGru software... Perhaps you should open issue or ask to @GrblGru on his site/forum...  
  
@++;  
Gauthier.

**Reply by Blackshuck4** [reply]
Hi Gauthier, 
First, many thanks for forwarding my query to GrblGru. For some reason, I thought you were using the Grblgru software, too. Appreciate your help, I can leave you in peace, now!

**Comment by GrblGru** [comment]
@Gauthier
Thanks for keeping me informed. :)

@Blackshuck4
It looks like there is still a bug in the GrblGru program.
I sent you a message in the CNC-Zone forum.

**Reply by Blackshuck4** [reply]
Thanks GrblGru - it's strange as I'm sure I had it all working! Thanks for sending me the message in CNC-Zone forum.
It really is a brilliant programme, and I'm so close to having a perfect 4-axis machine.
